File: dynamic_listings/lua/default/optimization/lazy_mode.lua
LazyObject = {}

function LazyObject:new(nums)
    local o = {
        numbers = nums
    }
    setmetatable(o, self)
    self.__index = self
    return o
end

function LazyObject:getObject(index)
    -- Calculates the halved number on-demand
    return self.numbers[index / 2];
end

File: dynamic_listings/python/default/optimization/lazy_mode.py
class LazyObject:
    list_reference: list[int] = None

    def __init__(self, lst: list[int]):
        # Saves the reference to the original list
        self.list_reference = lst

    def __getitem__(self, index: int):
        # Calculates the halved number on-demand
        return self.list_reference[index] / 2
